src/constraint_aware_append.c

```diff
@@ -54,18 +54,18 @@ static Plan *
 get_plans_for_exclusion(Plan *plan)
 {
 	/* Optimization: If we want to be able to prune */
-	/* when the node is a T_Result, then we need to peek */
+	/* when the node is a T_Result or T_Sort, then we need to peek */
 	/* into the subplans of this Result node. */
-	if (IsA(plan, Result))
+
+	switch (nodeTag(plan))
 	{
-		Result *res = (Result *) plan;
+		case T_Result:
+		case T_Sort:
+			return plan->lefttree;
 
-		if (res->plan.lefttree != NULL && res->plan.righttree == NULL)
-			return res->plan.lefttree;
-		if (res->plan.righttree != NULL && res->plan.lefttree == NULL)
-			return res->plan.righttree;
```

Comment:
The removed code suggests me that it would be nice to assert that `right tree` is `NULL`.

Reply:
I realised that the assertion is not needed, since it should not happen by the node constructions.